### ISSUE TITLE
Resource controllers do not respect group name

### ISSUE THREAD

**Issue by charles-rumley** [laravel/framework#14833]
Resource controllers use group prefixes instead of group names when generating names. Brief examination seems to show `ResourceRegistrar::getGroupResourceName()` uses `Router::getLastGroupPrefix()` instead of something like `Route::getLastGroupName()`.

Reproduced on Laravel v5.2.43 and v5.1.40.

This example from shows the difference between a non-resource route and a resource route. Route names shown are sourced from `RouteCollection::$nameList`.

```
Route::group(
    ['prefix' => 'group-prefix-1', 'as' => 'app.'],
    function () {

        // name: app.
        Route::get(
            '/',
            function () {
                return;
            }
        );

        // name: app.api.
        // expected
        Route::group(
            [
                "prefix" => "api",
                "as" => "api."
            ],
            function () {

                // name: app.api.get
                // expected
                Route::get('get', ['as' => 'get', 'uses' => 'Controller@get']);

                // name: app.api.group-prefix-1.api.resource.*
                // wait what?
                Route::resource('resource', 'ResourceController');
            }
        );
    }
);
```

**Comment by fernandobandeira** [comment]
This issue seems related with this one; #14547 maybe the same problem? Nice to have another example anyways.

**Comment by charles-rumley** [comment]
Yes, this issue duplicates #14547. Versions have been added.

**Comment by fernandobandeira** [comment]
This should be fixed in 5.3 since prefix isn't appending to the route name anymore.

**Comment by billmn** [comment]
@charles-rumley This issue can be closed?
